client/pd_service_discovery.go

```diff
@@ -493,39 +454,44 @@ func (c *pdBaseClient) updateFollowers(members []*pdpb.Member, leader *pdpb.Memb
 	c.followers.Store(addrs)
 }
 
-func (c *pdBaseClient) switchTSOAllocatorLeader(allocatorMap map[string]*pdpb.Member) error {
+func (c *pdServiceDiscovery) switchTSOAllocatorLeaders(allocatorMap map[string]*pdpb.Member) error {
 	if len(allocatorMap) == 0 {
 		return nil
 	}
+
+	allocMap := make(map[string]string)
 	// Switch to the new one
 	for dcLocation, member := range allocatorMap {
 		if len(member.GetClientUrls()) == 0 {
 			continue
 		}
-		addr := member.GetClientUrls()[0]
-		oldAddr, exist := c.GetTSOAllocatorServingAddrByDCLocation(dcLocation)
-		if exist && addr == oldAddr {
-			continue
```

Comment:
no need to compare them?

Reply:
need to compare them. Actually moved to the callback function updateTSOAllocatorServingAddrs, because pdServiceDiscovery can't call GetTSOAllocatorServingAddrByDCLocation() which is in the upper layer of tsoClient.

Reply:
func (c *tsoClient) updateTSOAllocatorServingAddrs(allocatorMap map[string]string) error {
         ...
	// Switch to the new one
	for dcLocation, addr := range allocatorMap {
                ...
	        oldAddr, exist := c.GetTSOAllocatorServingAddrByDCLocation(dcLocation)
		if exist && addr == oldAddr {
			continue
		}